### PR TITLE
Remove pragma in Charge Operations

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Charge Operation is now constructor initialized only
Lots of NotNull removed elsewhere

## References

* (https://github.com/Energinet-DataHub/geh-timeseries/issues/241)

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Point.cs [hunk 1]
```diff
@@ -21,6 +21,13 @@ namespace GreenEnergyHub.Charges.Domain.Charges
     /// </summary>
     public class Point
     {
+        public Point(int position, decimal price, Instant time)
+        {
+            Position = position;
+            Price = price;
+            Time = time;
+        }
+
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Point.cs:30]: Shouldn't you remove the setters while adding this ctor?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/ChargeOperationDto.cs [hunk 2]
```diff
@@ -29,62 +27,90 @@ namespace GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands
...
+                EndDateTime = endDateTime;
+                Points = points;
+            }
         }
 
         /// <summary>
         /// Contains a unique ID for the specific Charge Event, provided by the sender.
         /// </summary>
-        public string Id { get; set; }
+        public string Id { get; init; }
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/ChargeOperationDto.cs:66]: Does `init` make sense when all props are always set in ctor?

Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/ChargeOperationDto.cs]: removed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Point.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Point.cs:30]: Removed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Exceptions/ChargeOperationIsNullException.cs [hunk 3]
```diff
@@ -12,16 +12,11 @@
...
 
-using GreenEnergyHub.Charges.Domain.Charges;
-using GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands;
+using System;
 
-namespace GreenEnergyHub.Charges.Tests.Application.Charges.Validation.BusinessValidation
+namespace GreenEnergyHub.Charges.Infrastructure.Exceptions
 {
-    public class TestableChargeCommand : ChargeCommand
+    public class ChargeOperationIsNullException : Exception
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Exceptions/ChargeOperationIsNullException.cs:19]: Why do you want this specific exception type?

**Review by bemwen — APPROVED** [review]

**Pull request merged**